modules/tinydnsbackend/tinydnsbackend.cc

```diff
@@ -177,9 +179,14 @@ void TinyDNSBackend::getAllDomains(vector<DomainInfo> *domains, bool include_dis
       di.notified_serial = sd.serial;
       di.kind = DomainInfo::Master;
       di.last_check = time(0);
-      domains->push_back(di);
+      if (result.find(di) != result.end())
+        continue;
+      result.insert(di);
     }
   }
+
```

Comment:
Just a nit but we could call `domains->reserve(result.size())` as a small optimization :-)